src/unity/CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+# cmake file for ripple-libpp
+
+# Include this library in your own CMake project by including the following somewhere
+# below your project definition (`project(your_application)`)
+#
+# add_subdirectory(extras/ripple-libpp/src/unity)
+#
+# Customize the path as needed
+
+cmake_minimum_required(VERSION 3.1.0)
+
+list(APPEND CMAKE_MODULE_PATH
+  "${CMAKE_CURRENT_SOURCE_DIR}/../../extras/rippled/Builds/CMake")
+include(CMakeFuncs)
+
+include_directories(
+  ../../extras/rippled/src
+  ../../extras/rippled/src/beast/include
+  ../../extras/rippled/src/beast/extras)
+#  ../../extras/ripple-libpp/extras/rippled/src/srcp256k1
+#  ../../extras/ripple-libpp/extras/rippled/src/ed25519-donna
+
+determine_build_type()
+
+use_boost(
+  regex
+  system
+  thread)
+
+use_openssl(1.0.1)
+
+setup_build_boilerplate()
+
+if (WIN32 OR is_xcode)
+  # Rippled sources
+  foreach(curdir
+      beast/core
+      beast/hash
+      beast/utility
+      basics
+      core
+      crypto
+      json
+      protocol)
+    file(GLOB_RECURSE cursrcs ../../extras/rippled/src/ripple/${curdir}/*.h
+      ../../extras/rippled/src/ripple/${curdir}/*.cpp)
+    list(APPEND rippled_src "${cursrcs}")
+    list(APPEND non_unity_srcs "${cursrcs}")
+  endforeach()
+
+  file(GLOB_RECURSE all_headers src/*.hpp src/*.h
+    ../../extras/rippled/src/*.hpp ../../extras/rippled/src/*.h
+    ../../extras/*/CMakeFuncs.cmake)
```

Comment:
Yes, exactly. `all_headers` is only built for Win32 or XCode, and the files are flagged with the `HEADER_FILE_ONLY` property, so they get no implementation, so they won't build. (https://cmake.org/cmake/help/v3.6/prop_sf/HEADER_FILE_ONLY.html?highlight=header_file_only) Because of that, I can include files that are convenient to have in the IDE. Maybe I could use a better name, but I'm not going to bike shed it.